engine/minskyTensorOps.cc

```diff
@@ -499,7 +499,29 @@ namespace minsky
     }
     Timestamp timestamp() const override {return max(arg1->timestamp(), arg2->timestamp());}
     void setArguments(const TensorPtr& a1, const TensorPtr& a2) override {
-      arg1=a1; arg2=a2;
+	
+	// ensures that gather operation is x-vector aware, just like other tensor binops for ticket 1271	
```

Comment:
This won't work, because gather is not like other binops. The other binops are a generalisation of elementwise binary ops, whereas as gather is a generalisation of an index. The type (or units) of the second argument must match the type of first argument's x-vector. The InterpolateHC will come into play, but probably attached to the first argument.

Reply:
I understand, thank you. I have a few questions if you don't mind: will the implementation require modifications outside of minskyTensorOp.cc and operation.cc? Would it not be more appropriate to change the "Gather" op's base class to DimensionedArgCachedOp to achieve the desired result?

Reply:
I thought I'd responded to this. No -CachedOp is not approprate, as this each element's calculation will be independent of all of the others.

Reply:
Yes, you did, below. I haven't made any progress here and will probably have to go back to the drawing board on this one.. You can close it if you like.